diff --git a/dwm.c b/dwm.c
index 4465af1..4230e6b 100644
--- a/dwm.c
+++ b/dwm.c
@@ -430,17 +430,17 @@ buttonpress(XEvent *e)
 		focus(NULL);
 	}
 	if (ev->window == selmon->barwin) {
-		i = x = 0;
+		i = 0;
+		const int worksp_x = x = (m->ww - TEXTW(tags[0]) * 9) / 2;
+		const int lt_x = worksp_x - TEXTW(m->ltsymbol);
 		do
 			x += TEXTW(tags[i]);
 		while (ev->x >= x && ++i < LENGTH(tags));
-		if (i < LENGTH(tags)) {
+		if (i < LENGTH(tags) && ev->x >= worksp_x) {
 			click = ClkTagBar;
 			arg.ui = 1 << i;
-		} else if (ev->x < x + blw)
+		} else if (ev->x <= worksp_x && ev->x >= lt_x)
 			click = ClkLtSymbol;
-		else if (ev->x > selmon->ww - TEXTW(stext))
-			click = ClkStatusText;
 		else
 			click = ClkWinTitle;
 	} else if ((c = wintoclient(ev->window))) {
@@ -702,9 +702,10 @@ drawbar(Monitor *m)
 	Client *c;
 
 	/* draw status first so it can be overdrawn by tags later */
-	if (m == selmon) { /* status is only drawn on selected monitor */
-		drw_setscheme(drw, scheme[SchemeNorm]);
-		sw = TEXTW(stext) - lrpad + 2; /* 2px right padding */
+	/* status is only drawn on selected monitor */
+	if (m == selmon) {
+                drw_setscheme(drw, scheme[SchemeNorm]);
+		sw = TEXTW(stext) - lrpad;
 		drw_text(drw, m->ww - sw, 0, sw, bh, 0, stext, 0);
 	}
 
@@ -713,7 +714,9 @@ drawbar(Monitor *m)
 		if (c->isurgent)
 			urg |= c->tags;
 	}
-	x = 0;
+
+	/* for workspaces  */
+	const int worksp_x = x = (m->ww - TEXTW(tags[0]) * 9) / 2;
 	for (i = 0; i < LENGTH(tags); i++) {
 		w = TEXTW(tags[i]);
 		drw_setscheme(drw, scheme[m->tagset[m->seltags] & 1 << i ? SchemeSel : SchemeNorm]);
@@ -724,21 +727,22 @@ drawbar(Monitor *m)
 				urg & 1 << i);
 		x += w;
 	}
+
+	/* add blank space padding */
+	drw_setscheme(drw, scheme[SchemeNorm]);
+	drw_rect(drw, x, 0, m->ww - sw - x, bh, 1, 1);
+
+	/* for layout tile*/
 	w = blw = TEXTW(m->ltsymbol);
+	const int lt_x = x = worksp_x - w;
 	drw_setscheme(drw, scheme[SchemeNorm]);
-	x = drw_text(drw, x, 0, w, bh, lrpad / 2, m->ltsymbol, 0);
-
-	if ((w = m->ww - sw - x) > bh) {
-		if (m->sel) {
-			drw_setscheme(drw, scheme[m == selmon ? SchemeSel : SchemeNorm]);
-			drw_text(drw, x, 0, w, bh, lrpad / 2, m->sel->name, 0);
-			if (m->sel->isfloating)
-				drw_rect(drw, x + boxs, boxs, boxw, boxw, m->sel->isfixed, 0);
-		} else {
-			drw_setscheme(drw, scheme[SchemeNorm]);
-			drw_rect(drw, x, 0, w, bh, 1, 1);
-		}
-	}
+	drw_text(drw, x, 0, w, bh, lrpad / 2, m->ltsymbol, 0);
+
+	/* add blank space padding */
+	x = 0;
+	drw_setscheme(drw, scheme[SchemeNorm]);
+	drw_rect(drw, x, 0, lt_x, bh, 1, 1);
+
 	drw_map(drw, m->barwin, 0, 0, m->ww, bh);
 }
 
